tools/inference/lightning_inference.py

```diff
@@ -57,19 +57,17 @@ def infer(args: Namespace) -> None:
         visualization["image_save_path"] = args.output
 
     callbacks = None if not hasattr(args, "callbacks") else args.callbacks
-    engine = Engine(callbacks=callbacks, visualization=visualization)
+    engine = Engine(callbacks=callbacks, visualization=visualization, devices=1)
+    model = get_model(args.model)
 
     # create the dataset
     dataset = InferenceDataset(**args.data)
     dataloader = DataLoader(dataset)
 
-    # generate predictions
-    model = get_model(args.model)
     engine.predict(model=model, dataloaders=[dataloader], ckpt_path=args.ckpt_path)
 
 
 if __name__ == "__main__":
     parser = get_parser()
     config = parser.parse_args()
-    args = parser.instantiate_classes(config)
-    infer(args)
+    infer(config)
```

Comment:
I introduced it earlier because I was using the objects directly that were returned. But I changed the design so I reverted it. Plus, since we use the API in this method, it now instantiates the objects in the infer method. One of those lock step changes with API and CLI

Reply:
maybe we could still use `args`? It might look a bit more consistent with the name of the function `parse_args`? Thoughts?